server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -46,12 +47,35 @@ package object server {
       service => service.compose(AuthedRequest(authUser.run))
     }
 
-    /** TODO fs2 port -- replace |||
-    def apply[Err, T](authUser: Service[Request, Err \/ T], onFailure: Kleisli[Task, AuthedRequest[Err], MaybeResponse]): AuthMiddleware[T] = { service =>
-      (onFailure ||| service)
-        .local({authed: AuthedRequest[Either[Err, T]] => authed.authInfo.bimap(err => AuthedRequest(err, authed.req), suc => AuthedRequest(suc, authed.req))})
-        .compose(AuthedRequest(authUser.run))
+    implicit def serviceChoice: Choice[Service] = new Choice[Service] {
```

Comment:
I haven't tried compiling, but [you get a [`Choice[Kleisli[Task, ?, ?]]`](https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/data/Kleisli.scala#L121) if you have a `Monad[Task]`.    Can we remove this instance if we import `fs2.interop.cats._`?